src/test/test_iid_random_sequence.py

```diff
@@ -0,0 +1,81 @@
+# numpydoc ignore=GL08
+
+import jax.numpy as jnp
+import numpyro
+import numpyro.distributions as dist
+import pytest
+from scipy.stats import kstest
+
+from pyrenew.metaclass import (
+    DistributionalRV,
+    SampledValue,
+    StaticDistributionalRV,
+)
+from pyrenew.process import IIDRandomSequence, StandardNormalSequence
+
+
+@pytest.mark.parametrize(
+    ["distribution", "n"],
+    [
+        [dist.Normal(0, 1), 1000],
+        [dist.Cauchy(2, 325.0), 13532],
+        [dist.Normal(jnp.array([2.0, 3.0, -5.235]), 0.25), 622],
+    ],
+)
+def test_iidrandomsequence_with_dist_rv(distribution, n):
+    """
+    Check that an IIDRandomSequence can be
+    initialized and sampled from when the element_rv is
+    a distributional RV, including with array-valued
+    distributions
+    """
+    element_rv = DistributionalRV("el_rv", distribution=distribution)
+    rseq = IIDRandomSequence(name="randseq", element_rv=element_rv)
+    expected_shape = distribution.batch_shape
```

Comment:
since `expected_shape` is always changed in the `if-else` block below, I think a different name or just re-using `distribution.batch_shape` below is more clear.

Reply:
fixed in e84f975